### DISCUSSION TITLE
SHIFT key for TerminalAccessory

### DISCUSSION THREAD

**Discussion by alexeyhatkevich** [migueldeicaza/SwiftTerm#196] [Ideas]
What about adding SHIFT key support?
It will be great to insert and enhance this function:
func applyShiftToEventCharacters (_ ch: String) -> [UInt8]
    {
        let arr = [UInt8](ch.utf8)
        if arr.count == 1 {
            let ch = Character(UnicodeScalar (arr [0]))
            var value: UInt8
            switch ch {
            case "a"..."z":
                value = (ch.asciiValue! - 0x20)
            default:
                value = (ch.asciiValue!)
            }
            return [value]
        }
        return []
    }

**Comment by alexeyhatkevich** [comment]
Hi @migueldeicaza. Any ideas?

**Comment by migueldeicaza** [comment]
Hello,

I am not sure I understand, can you tell me more about what seems to be missing and where?

**Reply by alexeyhatkevich** [reply]
Add SHIFT button into inputAccessoryView

**Reply by alexeyhatkevich** [reply]
And sticking mode for ctrl and shift buttons

**Reply by migueldeicaza** [reply]
Yes, but I don't understand what the shift would be for.  There is already one on the main keyboard.
